crates/net/discv5/src/config.rs

```diff
@@ -165,21 +205,22 @@ impl DiscV5ConfigBuilder {
             tcp_port,
             other_enr_data,
             allow_no_tcp_discovered_nodes,
-            self_lookup_interval: lookup_interval,
+            self_lookup_interval,
+            filter_discovered_peer,
         }
     }
 }
 
 /// Config used to bootstrap [`discv5::Discv5`].
 #[derive(Debug)]
-pub struct DiscV5Config {
+pub struct DiscV5Config<Filter = DefaultFilter> {
     /// Config used by [`discv5::Discv5`]. Contains the [`ListenConfig`], with the discovery listen
     /// socket.
     pub(super) discv5_config: discv5::Config,
     /// Nodes to boot from.
     pub(super) bootstrap_nodes: HashSet<BootNode>,
     /// [`ForkId`] to set in local node record.
-    pub(super) fork_id: Option<ForkId>,
+    pub(super) fork_id: (&'static [u8], ForkId),
```

Comment:
is the forkId manadatory?
because if not configured properly we currently end up with MAINNET

Reply:
isn't L1 mainnet not a natural default for ethereum? it's the same thing that happens if you start a reth node from cli without specifying the chain, so I think it holds.